src/pnpq/devices/waveplate_stub.py

```diff
@@ -0,0 +1,217 @@
+import logging
+import time
+
+from pnpq.errors import (
+    DevicePortNotFoundError,
+    DeviceDisconnectedError,
+    WavePlateMoveNotCompleted,
+    WavePlateHomedNotCompleted,
+    WavePlateGetPosNotCompleted,
+    WavePlateCustomRotateError,
+    WaveplateInvalidStepsError,
+    WaveplateInvalidDegreeError,
+    WaveplateEnableChannelError,
+    WaveplateInvalidMotorChannelError,
+)
+
+class WaveplateStub:
+    """Stub Waveplate Device Class"""
+    resolution: int
+    max_steps: int
+    relative_home: float
+
+    def __init__(self):
+        # Stub Serial Number
+        # TODO: Custom serial number from initializer
+        self.device_sn = "stubwaveplate"
+
+        self.resolution = 136533
+        self.max_steps = 136533
+        self.rotate_timeout = 10
+        self.home_timeout = 20
+        self.max_channel = 1
+        self.auto_update = False
```

Comment:
1. This isn't really the right way to define variables in Python, I think. The variables on 19-21 are static, and I don't think there's ever a value being assigned to them. The variables with the same names on lines 28-33 are actually totally separate. See [this StackOverflow discussion](https://stackoverflow.com/questions/9056957/correct-way-to-define-class-variables-in-python) for more information.
2. That being said, it would be a good idea to provide type information for the `self` variables.
3. Also, it would be a VERY good idea to add comments explaining what these variables actually mean and how they relate to each other. What is "resolution?" How do you calculate degrees from it? What are the timeouts?
4. It might be a good idea to implement a method that calculates and returns the current absolute position in degrees. This might follow the Python getter pattern.

Reply:
I agree with 1 - 3, but I think there may be an issue with 4 if we use Python's getter because on the real device, `getpos()` could throw errors. Maybe it will just be in the form of another function which calls the `getpos()` function.

Reply:
> here may be an issue with 4 if we use Python's getter because on the real device, `getpos()` could throw errors

We might be talking about different patterns here. I'm thinking about a function.